core/src/main/java/bio/terra/pearl/core/dao/BaseJdbiDao.java

```diff
@@ -205,6 +205,11 @@ public List<T> findAll(List<UUID> uuids) {
         return findAllByPropertyCollection("id", uuids);
     }
 
+    /** finds all the entities by the given ids, sorted in the same order as the ids */
+    public List<T> findAllPreserveOrder(List<UUID> uuids) {
```

Comment:
It wasn't introduced in this PR, but since it's being used, I think it would be worth throwing an exception in the underlying `findAllByPropertyCollectionPreserveOrder` method if the length of the supplied `columnValues` doesn't match the number of rows that end up getting returned. It seems easy to misuse it, and I could imagine some very nasty bugs if the returned number of rows didn't match what the caller expected